### PR TITLE
[Feature] Add edited ArticleReproduction_CV.md 

### PR DESCRIPTION
**Pull request by shiyutang**

### CODE REVIEW

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md [hunk 1]
```diff
@@ -0,0 +1,1048 @@
...
+<a name="2"></a>
+## 2. 整体框图
+
+<a name="2.1"></a>
+### 2.1 流程概览
+
+面对一篇计算机视觉论文，复现该论文的整体流程如下图所示：
+
+<div align="center">
+<img src="images/framework_reprodcv.png"  width = "800" />
```
Comment (littletomatodonkey) [inline tutorials/tipc/train_infer_python/ArticleReproduction_CV.md:71]: 这张图片添加一下吧，根据之前提供的ppt修改下框图，重新画一个提交就好

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md [hunk 2]
```diff
@@ -0,0 +1,1048 @@
...
+* 存取指定节点的输入输出tensor
+* 基于文件的tensor读写
+* 2个字典的对比验证
+* 对比结果的输出与记录
+
+更多API与使用方法可以参考：[reprod_log API使用说明](https://github.com/WenmuZhou/reprod_log/blob/master/README.md)。
+
+#### 2.2.2 reprod_log使用demo
+
+下面基于[示例代码](https://github.com/littletomatodonkey/AlexNet-Prod/tree/master/pipeline/reprod_log_demo)，给出如何使用该工具。
```
Comment (littletomatodonkey) [inline tutorials/tipc/train_infer_python/ArticleReproduction_CV.md:96]: 都改成mv3的示例代码地址，另外就是关于mv3复现部分的内容都给成相对路径的格式

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md [hunk 3]
```diff
@@ -0,0 +1,1048 @@
...
+
+**【实战】**
+
+MobilnetV3网络结构的PyTorch实现: [mobilenetv3-pytorch](TODO)
+
+对应转换后的PaddlePaddle实现: [mobilenetv3-paddle](TODO)
+
+**【FAQ】**
+
+1. 遇到 paddle 不支持的API怎么办？
```
Comment (littletomatodonkey) [inline tutorials/tipc/train_infer_python/ArticleReproduction_CV.md]: 标号改成（1）（2）之类的吧，以此类推

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md [hunk 4]
```diff
@@ -0,0 +1,1048 @@
...
+
+    1. 进一步参考[API映射表](https://www.paddlepaddle.org.cn/documentation/docs/zh/guides/08_api_mapping/pytorch_api_mapping_cn.html) ：由于PaddlePaddle与PyTorch对于不同名称的API，实现的功能可能是相同的，比如[paddle.optimizer.lr.StepDecay](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/optimizer/lr/StepDecay_cn.html#stepdecay)与[torch.optim.lr_scheduler.StepLR](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.StepLR.html#torch.optim.lr_scheduler.StepLR) ，因此需要进一步确认当前的确API没有实现。
```
Comment (littletomatodonkey) [inline tutorials/tipc/train_infer_python/ArticleReproduction_CV.md:186]: 这里建议写成a，b，c等，不然和上面的标号有歧义

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md [hunk 5]
```diff
@@ -0,0 +1,1048 @@
...
+    ```python
+      torch.stack([
+        per_locations[:, 0] - per_box_regression[:, 0],
+        per_locations[:, 1] - per_box_regression[:, 1],
+        per_locations[:, 0] + per_box_regression[:, 2],
+        per_locations[:, 1] + per_box_regression[:, 3],
+      ], dim=1)
+    ```
+    3. 尝试自己开发算子：我们非常欢迎开发者向我们贡献代码：）。
+    4. 在[这里](https://github.com/PaddlePaddle/Paddle/issues/new/choose)向paddle 提交issue：列出Paddle不支持的实现，开发人员会根据优先级进行开发。
```
Comment (littletomatodonkey) [inline tutorials/tipc/train_infer_python/ArticleReproduction_CV.md:189]: 说明下什么情况可以提卡片，比如说: 如果您不希望或者暂时没有时间开发新的算子，可以向paddle提交issue之类的。

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md [hunk 6]
```diff
@@ -0,0 +1,1048 @@
...
+    6. 得知API开发完成之后，安装编译环境：
```
Comment (littletomatodonkey) [inline tutorials/tipc/train_infer_python/ArticleReproduction_CV.md]: 1，2，3，4，6，这里的顺序看着有点问题

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md (hunk 1 above)
Comment (shiyutang) [reply tutorials/tipc/train_infer_python/ArticleReproduction_CV.md:71]: done

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md (hunk 2 above)
Comment (shiyutang) [reply tutorials/tipc/train_infer_python/ArticleReproduction_CV.md:96]: 这个部分reprod_log 的代码没有单独抽出来，因此就使用之前的链接~

Comment (shiyutang) [reply tutorials/tipc/train_infer_python/ArticleReproduction_CV.md:96]: > 另外就是关于mv3复现部分的内容都给成相对路径的格式
好的，我后面添加链接的时候，说明是哪个文件~

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md (hunk 3 above)
Comment (shiyutang) [reply tutorials/tipc/train_infer_python/ArticleReproduction_CV.md]: （1） 应该不是markdown 序号的标识符，改了之后看着格式有点奇怪

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md (hunk 4 above)
Comment (shiyutang) [reply tutorials/tipc/train_infer_python/ArticleReproduction_CV.md:186]: 这里也是，如果改了，就不是有序列表了，所以就格式会混在一起

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md (hunk 5 above)
Comment (shiyutang) [reply tutorials/tipc/train_infer_python/ArticleReproduction_CV.md:189]: done

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md (hunk 6 above)
Comment (shiyutang) [reply tutorials/tipc/train_infer_python/ArticleReproduction_CV.md]: done

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md [hunk 7]
```diff
@@ -0,0 +1,1046 @@
...
+MobilnetV3网络结构的PyTorch实现: [mobilenetv3_prod/Step1-5/mobilenetv3_ref/torchvision/models/mobilenet_v3_torch.py](https://github.com/PaddlePaddle/models/blob/release%2F2.2/tutorials/mobilenetv3_prod/Step1-5/mobilenetv3_ref/torchvision/models/mobilenet_v3_torch.py)
+
+对应转换后的PaddlePaddle实现: [mobilenetv3_prod/Step1-5/mobilenetv3_paddle/paddlevision/models/mobilenet_v3_paddle.py](https://github.com/PaddlePaddle/models/blob/release%2F2.2/tutorials/mobilenetv3_prod/Step1-5/mobilenetv3_paddle/paddlevision/models/mobilenet_v3_paddle.py)
+
+**【FAQ】**
+
+- 遇到 paddle 不支持的API怎么办？
+
+    1. 进一步参考[API映射表](https://www.paddlepaddle.org.cn/documentation/docs/zh/guides/08_api_mapping/pytorch_api_mapping_cn.html) ：由于PaddlePaddle与PyTorch对于不同名称的API，实现的功能可能是相同的，比如[paddle.optimizer.lr.StepDecay](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/optimizer/lr/StepDecay_cn.html#stepdecay)与[torch.optim.lr_scheduler.StepLR](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.StepLR.html#torch.optim.lr_scheduler.StepLR) ，因此需要进一步确认当前的确API没有实现。
+    2. 尝试使用替代实现进行复现：如下面的PyTorch代码，PaddlePaddle中可以通过slice + concat API的组合形式进行功能实现。在第二个问题，我们还可以看到其他一些torch API可用 PaddlePaddle API 实现。
```
Comment (littletomatodonkey) [inline tutorials/tipc/train_infer_python/ArticleReproduction_CV.md]: 这个第2点去掉吧，目前tensor已经支持下标索引了

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md [hunk 8]
```diff
@@ -0,0 +1,1046 @@
...
+本部分对齐建议对照[PaddlePaddle优化器API文档](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/optimizer/Overview_cn.html)、[PaddlePaddle正则化API文档](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/regularizer/L2Decay_cn.html)与参考代码的优化器实现进行对齐，用之后的反向对齐统一验证该模块的正确性。另外在学习率中还可以参考学习率对齐代码[Step1-5/05_test_backward.py#L23](https://github.com/PaddlePaddle/models/blob/release%2F2.2/tutorials/mobilenetv3_prod/Step1-5/05_test_backward.py#L23)。
+
+**【FAQ】**
+- 怎么实现参数分组学习率策略？
+
+    * Paddle目前支持在 `optimizer` 中通过设置 `params_groups` 的方式设置不同参数的更新方式，可以参考[代码示例](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/optimizer/optimizer.py#L107) 。
+
+- 有没有什么其他影响优化不对齐的原因？
+    * 有些模型训练时，会使用梯度累加策略，即累加到一定step数量之后才进行参数更新，这时在实现上需要注意对齐。
+    * 在图像分类领域，大多数Vision Transformer模型都采用了AdamW优化器，并且会设置weigh decay，同时部分参数设置为no weight decay，例如位置编码的参数通常设置为no weight decay，no weight decay参数设置不正确，最终会有明显的精度损失，需要特别注意。一般可以通过分析模型权重来发现该问题，分别计算官方模型和复现模型每层参数权重的平均值、方差，对每一层依次对比，有显著差异的层可能存在问题，因为在weight decay的作用下，参数权重数值会相对较小，而未正确设置no weight decay，则会造成该层参数权重数值异常偏小。设置no weight decay 可以参照[这里](https://github.com/PaddlePaddle/PaddleClas/blob/release%2F2.3/ppcls/arch/backbone/model_zoo/resnest.py#L72)。
```
Comment (littletomatodonkey) [inline tutorials/tipc/train_infer_python/ArticleReproduction_CV.md]: `weigh decay` -> `weight decay`

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md [hunk 9]
```diff
@@ -0,0 +1,1046 @@
...
+    # Define the linear layer.
+    m = paddle.nn.Linear(2, 4)
+    print(m.bias)
+    if isinstance(m, nn.Layer):
+        print("set m.bias")
+        m.bias.set_value(np.ones(shape=m.bias.shape, dtype="float32"))
+        print(m.bias)
+    ```
+
+3. 初始化是怎么影响不同类型的模型的？
```
Comment (littletomatodonkey) [inline tutorials/tipc/train_infer_python/ArticleReproduction_CV.md]: 这块和上面一样，改成`-`吧，

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md [hunk 10]
```diff
@@ -0,0 +1,1046 @@
...
+
+* 【强烈】建议先做完反向对齐之后再进行模型训练对齐，二者之间的不确定量包括：数据集、PaddlePaddle与参考代码在模型training mode下的区别，初始化参数。
+* 在训练对齐过程中，受到较多随机量的影响，精度有少量diff是正常的，以ImageNet1k数据集的分类为例，diff在0.15%以内可以认为是正常的，这里可以根据不同的任务，适当调整对齐检查的阈值(`ReprodDiffHelper.report`函数中的`diff_threshold`参数)。
+* 训练过程中的波动是正常的，如果最终收敛结果不一致，可以
+    * 仔细排查Dropout、BatchNorm以及其他组网模块及超参是否无误。
+    * 基于参考代码随机生成一份预训练模型，转化为PaddlePaddle的模型，并使用PaddlePaddle加载训练，对比二者的收敛曲线与最终结果，排查初始化影响。
+    * 使用参考代码的Dataloader生成的数据，进行模型训练，排查train dataloader的影响。
+
+**【实战】**
+【todo】
```
Comment (littletomatodonkey) [inline tutorials/tipc/train_infer_python/ArticleReproduction_CV.md]: 这个todo是需要补充什么内容呢？

参考代码训练log：https://paddle-model-ecology.bj.bcebos.com/model/mobilenetv3_reprod/train_mobilenet_v3_small_ref.log

paddle训练log：https://paddle-model-ecology.bj.bcebos.com/model/mobilenetv3_reprod/train_mobilenet_v3_small.log

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md (hunk 8 above)
Comment (shiyutang) [reply tutorials/tipc/train_infer_python/ArticleReproduction_CV.md]: done

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md (hunk 7 above)
Comment (shiyutang) [reply tutorials/tipc/train_infer_python/ArticleReproduction_CV.md]: 使用其他例子替代

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md (hunk 9 above)
Comment (shiyutang) [reply tutorials/tipc/train_infer_python/ArticleReproduction_CV.md]: done

File: tutorials/tipc/train_infer_python/ArticleReproduction_CV.md (hunk 10 above)
Comment (shiyutang) [reply tutorials/tipc/train_infer_python/ArticleReproduction_CV.md]: 增加训练log，去除todo

**Review by littletomatodonkey — APPROVED** [review]
